zigzag.go

```diff
@@ -0,0 +1,133 @@
+package hdrhistogram
+
+import "fmt"
+
+const truncatedErrStr = "Truncated compressed histogram decode. Expected minimum length of %d bytes and got %d."
+
+// Read an LEB128 ZigZag encoded long value from the given buffer
+func zig_zag_decode_i64(buf []byte) (signedValue int64, n int, err error) {
+	buflen := len(buf)
+	if buflen < 1 {
+		return 0, 0, nil
+	}
+	var value = uint64(buf[0]) & 0x7f
+	n = 1
+	if (buf[0] & 0x80) != 0 {
+		if buflen < 2 {
+			err = fmt.Errorf(truncatedErrStr, 2, buflen)
+			return
+		}
+		value |= uint64(buf[1]) & 0x7f << 7
+		n = 2
+		if (buf[1] & 0x80) != 0 {
+			if buflen < 3 {
+				err = fmt.Errorf(truncatedErrStr, 3, buflen)
+				return
+			}
+			value |= uint64(buf[2]) & 0x7f << 14
+			n = 3
+			if (buf[2] & 0x80) != 0 {
+				if buflen < 4 {
+					err = fmt.Errorf(truncatedErrStr, 4, buflen)
+					return
+				}
+				value |= uint64(buf[3]) & 0x7f << 21
+				n = 4
+				if (buf[3] & 0x80) != 0 {
+					if buflen < 5 {
+						err = fmt.Errorf(truncatedErrStr, 5, buflen)
+						return
+					}
+					value |= uint64(buf[4]) & 0x7f << 28
+					n = 5
+					if (buf[4] & 0x80) != 0 {
+						if buflen < 6 {
+							err = fmt.Errorf(truncatedErrStr, 6, buflen)
+							return
+						}
+						value |= uint64(buf[5]) & 0x7f << 35
+						n = 6
+						if (buf[5] & 0x80) != 0 {
+							if buflen < 7 {
+								err = fmt.Errorf(truncatedErrStr, 6, buflen)
```

Comment:
looks like some copy pasting of the fmt.Errorf lines, and the numbers after 6 aren't right (should be 7,8,9 I think)